classes/01class/exercises/c01-git04/danielvca/README.md

```diff
@@ -0,0 +1,39 @@
+﻿
+
+# c01-git04
+
+## Questions (answer inline)
+
+1. Let's suppose you remotely pushed the `my_env.txt` file above. A colleague asks you to remove this information from Git. What's your colleague worried about?
+
+> We`re not supposed to store sensitive information in our repo. We can
+> also leverage services like key_vault to store this kind of info.
+
+2. If you modify the file in your workspace, then commit and push it, will it be enough to erase this password information from the repository? (It's not). Why?
+
+> No, Git object store would still have a copy of the said file.
+
+3. If you delete the file and push it, then create a new one with the rest of the information, is it enough? (It's not). Why?
+
+> No. Sensitive information can still be retrieved as the history would
+> still be accessible.
+
+4. How to fix this? How do you remove something from Git history when it is in the remote repository?
+
+> We can either use bfg or a combination of git rebase and git
+> filter-branch
```

Comment:
The new recommended way of solving this is by using `git filter-repo`, but `filter-branch` can still be used.
https://github.com/newren/git-filter-repo